packages/task/src/browser/task-configurations.ts

```diff
@@ -85,41 +78,32 @@ export class TaskConfigurations implements Disposable {
     @inject(ProvidedTaskConfigurations)
     protected readonly providedTaskConfigurations: ProvidedTaskConfigurations;
 
-    constructor(
-        @inject(FileSystemWatcher) protected readonly watcherServer: FileSystemWatcher,
-        @inject(FileSystem) protected readonly fileSystem: FileSystem
-    ) {
-        this.toDispose.push(watcherServer);
-        this.toDispose.push(
-            this.watcherServer.onFilesChanged(async changes => {
-                try {
-                    const watchedConfigFileChanges = changes.filter(change =>
-                        this.watchedConfigFileUris.some(fileUri => FileChangeEvent.isAffected([change], new URI(fileUri)))
-                    ).map(relatedChange => (
-                        { uri: relatedChange.uri.toString(), type: relatedChange.type }
-                    ));
-                    if (watchedConfigFileChanges.length >= 0) {
-                        await this.onDidTaskFileChange(watchedConfigFileChanges);
-                        if (this.client) {
-                            this.client.taskConfigurationChanged(this.getTaskLabels());
-                        }
-                    }
-                } catch (err) {
-                    console.error(err);
-                }
-            })
-        );
+    @inject(TaskConfigurationManager)
+    protected readonly taskConfigurationManager: TaskConfigurationManager;
+
+    constructor() {
         this.toDispose.push(Disposable.create(() => {
             this.tasksMap.clear();
             this.taskCustomizationMap.clear();
-            this.watchersMap.clear();
             this.rawTaskConfigurations.clear();
             this.client = undefined;
         }));
     }
 
     @postConstruct()
     protected init(): void {
+        this.toDispose.push(
+            this.taskConfigurationManager.onChangedTaskConfig(async change => {
+                try {
+                    await this.onDidTaskFileChange([change]);
```

Comment:
looking at the implementation, strange that `removeTasks` is exposed, it should be called by clients since it does not change underlying files

the same about `removeTaskCustomizations`

Reply:
i will change them to `private`. Thank you !